dbcrossbarlib/src/drivers/postgres_shared/column.rs

```diff
@@ -49,17 +60,36 @@ impl PgColumn {
                 dimension_count,
                 ty: PgScalarDataType::Bigint,
             } => {
-                if *dimension_count != 1 {
-                    return Err(format_err!(
-                        "cannot output bigint[] columns with dimension > 1"
-                    ));
-                }
+                check_dimension(*dimension_count)?;
                 write!(
                     f,
                     r#"(SELECT array_to_json(array_agg((elem))) FROM (SELECT elem::text FROM unnest({name}) AS elem) AS elems) AS {name}"#,
                     name = name,
                 )?;
             }
+            // Format timestamps as ISO 8601 instead of Postgres's format.
+            PgDataType::Array {
+                dimension_count,
+                ty: PgScalarDataType::TimestampWithoutTimeZone,
+            } => {
+                check_dimension(*dimension_count)?;
+                write!(
+                    f,
+                    r#"(SELECT array_to_json(array_agg((elem))) FROM (SELECT TRIM(TRAILING '.' FROM TRIM(TRAILING '0' FROM to_char(elem, 'YYYY-MM-DD"T"HH24:MI:SS.US'))) AS elem FROM unnest({name}) AS elem) AS elems) AS {name}"#,
```

Comment:
omg

![don't look inside](https://cdn.instructables.com/FV2/A2B3/GFMC84KY/FV2A2B3GFMC84KY.LARGE.jpg)

Reply:
And that's only the second-worst! The worst has an extra `at time zone 'UTC'` and `|| 'Z'`.

But Redshift doesn't allow us to define temporary functions, and we probably shouldn't assume that we have the necessary permissions to define them even on PostgreSQL. So we wind up with some pretty funky expressions in our export `SELECT`.

I can see two ways to improve this code:

1. Refactor the array types so that the "array SQL" is generated separately from the "data serialization SQL". This would clean things up slightly, at the cost of making other things slightly messier.
2. Set up some way to use whitespace and comments in the SQL source, but strip them from the generated SQL.

Honestly, I really love the way BigQuery permits us to define temporary functions inside a single query. It's a purely syntactic thing, but it simplifies this sort of code tremendously.